### PR TITLE
Support input packages

### PR DESCRIPTION
**Pull request by mtojek**
Meta-issue: https://github.com/elastic/package-spec/issues/319

Blocker: https://github.com/elastic/package-spec/pull/328

This PR modifies the Package Registry to support "input packages".

Changes:
1. Use `sql_input` as test package.
2. Add missing fields to "package index" API method.
3. Filter by "type" with /search API method.
4. Regenerate expected test results.

### CODE REVIEW

File: testdata/generated/search-all.json
Comment (jsoriano) [inline testdata/generated/search-all.json:363]: What are old versions of Kibana do with packages that are not integrations?

Should we filter by `type: integration` when no `type` is passed? and maybe have a `type: all` to list all packages.

Or maybe this is a point where we should start versioning the API?

File: main_test.go
Comment (jsoriano) [inline main_test.go:89]: Maybe we should add test cases for:
* An input package with multiple versions.
* A package that has a type on one version, and another type on another version.

File: testdata/generated/search-all.json
Comment (mtojek) [reply testdata/generated/search-all.json:363]: > and maybe have a type: all to list all packages.

I wouldn't like to design an API that won't be used by Fleet at all (KISS).

> Should we filter by type: integration when no type is passed? 

That would be counter-intuitive, you want to search for packages, but for some historical reason, the /search filters some of them. I would only go with that option to prevent any problems on the Fleet side.

> Or maybe this is a point where we should start versioning the API?

It's always a good practice to version it, but not sure if this is a "dead end" to introduce versioning or if we just enable another GET parameter.

Comment (jsoriano) [reply testdata/generated/search-all.json:363]: Well, then going back to the first question, what are current/old versions of Kibana do with packages that are not integrations?

Comment (mtojek) [reply testdata/generated/search-all.json:363]: Passing the question to @joshdover.

File: main_test.go
Comment (mtojek) [reply main_test.go:89]: > An input package with multiple versions.

I added one more revision of `sql_input`.

> A package that has a type on one version, and another type on another version.

`integration_input` package, older revision is integration, newer is input.

I haven't noticed anything unusual, but feel free to review API responses in case I missed something.

**Review by jsoriano — COMMENTED** [review]
Looks good, waiting for the discussion about backwards compatibility with Kibana/Fleet.

File: main_test.go
Comment (mtojek) [reply main_test.go:89]: Good idea, fixed

**Comment by mtojek** [comment]
As the input-type spec has been [released](https://github.com/elastic/package-spec/pull/328), I suppose that this is the next task to be reviewed/merged.

File: testdata/generated/search-all.json
Comment (joshdover) [reply testdata/generated/search-all.json]: This seems like something that could easily be tested manually without code-level knowledge of Fleet. That would be how I test this personally.

**Review by jlind23 — APPROVED** [review]

**Review by jlind23 — COMMENTED** [review]
LGTM

File: testdata/generated/search-all.json
Comment (mtojek) [reply testdata/generated/search-all.json]: Ok, I started the 7.16.3 stack and was able to list the `sql_input` package (`type: input`), which is not good. On the other hand if we use Kibana constraint to cut old Kibana versions that should do the trick.

Zrzut ekranu 2022-05-26 o 13 46 20

WDYT @joshdover? I can go any direction, either adjust API or adjust Kibana constraints.

Comment (jsoriano) [reply testdata/generated/search-all.json]: > On the other hand if we use Kibana constraint to cut old Kibana versions that should do the trick.

Agree that this can be a good solution. We should probably add a check for this in the package-spec, and we would need to wait to know the version of Kibana that is going to support input packages for this.

Comment (mtojek) [reply testdata/generated/search-all.json]: Sounds like a good idea, but, to be honest, I wouldn't like to block this PR until that happens.

Comment (jsoriano) [reply testdata/generated/search-all.json]: Ah yeah, I didn't want to mean that :slightly_smiling_face: But this may delay declaring input package as GA in the spec.

**Review by jsoriano — APPROVED** [review]

File: testdata/generated/search-all.json
Comment (jlind23) [reply testdata/generated/search-all.json]: @mtojek I would say that each new input package should start with a kibana version constraints higher or equals to 8.3 which should then do the trick isn't it?

Comment (mtojek) [reply testdata/generated/search-all.json]: Yes, Julien, that's the idea. Just waiting for the confirmation/approval from @joshdover.

Comment (joshdover) [reply testdata/generated/search-all.json]: Kibana version constraint enforced by package-spec SGTM, agreed it can be done as a follow up.

**Pull request merged**
